harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -374,9 +374,7 @@ def _auto_step_lr_scheduler_per_batch(
     def _should_update_scaler(self) -> bool:
         if not self.context._scaler or not self.context.experimental._auto_amp:
             return False
-        if self.context.distributed.size > 1:
-            return self.context._should_communicate_and_update()  # type: ignore
-        return True
+        return self.context._should_communicate_and_update()  # type: ignore
```

Comment:
I think this alternative is better for being a more conservative change; also, the reason for it is easier to understand just by looking at the diff. 
If there's any possibility of `_should_communicate_and_update` causing an error (in the sense of misuse) if neither `distributed.size > 1` nor `_aggregation_frequency > 1` are `True`, this will prevent that.